datalake/src/main/java/com/sequenceiq/datalake/flow/verticalscale/diskupdate/handler/DatalakeDiskUpdateHandler.java

```diff
@@ -0,0 +1,78 @@
+package com.sequenceiq.datalake.flow.verticalscale.diskupdate.handler;
+
+import static com.sequenceiq.datalake.flow.verticalscale.diskupdate.event.DatalakeDiskUpdateStateSelectors.DATALAKE_DISK_UPDATE_HANDLER_EVENT;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.DatalakeV4Endpoint;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.StackV4Endpoint;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.SdxDiskUpdateRequest;
+import com.sequenceiq.cloudbreak.eventbus.Event;
+import com.sequenceiq.cloudbreak.eventbus.Promise;
+import com.sequenceiq.datalake.entity.DatalakeStatusEnum;
+import com.sequenceiq.datalake.flow.verticalscale.diskupdate.event.DatalakeDiskUpdateEvent;
+import com.sequenceiq.datalake.flow.verticalscale.diskupdate.event.DatalakeDiskUpdateFailedEvent;
+import com.sequenceiq.datalake.flow.verticalscale.diskupdate.event.DatalakeDiskUpdateStateSelectors;
+import com.sequenceiq.flow.reactor.api.event.EventSender;
+import com.sequenceiq.flow.reactor.api.handler.EventSenderAwareHandler;
+
+@Component
+public class DatalakeDiskUpdateHandler extends EventSenderAwareHandler<DatalakeDiskUpdateEvent> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatalakeDiskUpdateHandler.class);
+
+    @Inject
+    private StackV4Endpoint stackV4Endpoint;
+
+    @Inject
+    private DatalakeV4Endpoint datalakeV4Endpoint;
+
+    public DatalakeDiskUpdateHandler(EventSender eventSender) {
+        super(eventSender);
+    }
+
+    @Override
+    public String selector() {
+        return DATALAKE_DISK_UPDATE_HANDLER_EVENT.selector();
+    }
+
+    @Override
+    public void accept(Event<DatalakeDiskUpdateEvent> datalakeDiskUpdateEventEvent) {
+        LOGGER.debug("In DatalakeDiskUpdateHandler.accept");
+        DatalakeDiskUpdateEvent payload = datalakeDiskUpdateEventEvent.getData();
+        try {
+            Long stackId = payload.getDatalakeDiskUpdateRequest().getStackId();
+            LOGGER.debug("Starting Disk Update for datalake.");
+            SdxDiskUpdateRequest sdxDiskUpdateRequest = new SdxDiskUpdateRequest();
+            sdxDiskUpdateRequest.setVolumesToUpdate(payload.getVolumesToBeUpdated());
+            sdxDiskUpdateRequest.setCloudPlatform(payload.getCloudPlatform());
+            sdxDiskUpdateRequest.setDiskUpdateRequest(payload.getDatalakeDiskUpdateRequest());
+            datalakeV4Endpoint.updateDiskTypeAndSize(sdxDiskUpdateRequest);
+            LOGGER.debug("Resizing Disks for datalake using orchestrator.");
+            datalakeV4Endpoint.resizeDisks(stackId, payload.getDatalakeDiskUpdateRequest().getGroup());
```

Comment:
core side implementation should be in flow and in sdx service you should poll core service regarding flow in another flow step

Reply:
@horadla23 please elaborate. i am not sure i understand

Reply:
pls check any code in datalake service. when manipulating sdx, we call datalake API in datalake service, calling the related API of core from a flow and then polling it until it finishes. There are a lot of examples in datalake service.
One example: 
https://github.com/hortonworks/cloudbreak/blob/af8e7a4af65df3e00e403d0ba46b5b8976512604/datalake/src/main/java/com/sequenceiq/datalake/flow/start/SdxStartActions.java#LL94C33-L94C33
https://github.com/hortonworks/cloudbreak/blob/master/datalake/src/main/java/com/sequenceiq/datalake/flow/start/handler/SdxStartWaitHandler.java